### PR TITLE
E2E: Refactor Mini Cart to be ready for fully parallel

### PR DESCRIPTION
**Pull request by dinhtungdu**
## What

This PR:
- refactors the Mini Cart test to remove the usage of CSS selector in favor of role based locators.
- makes the tests fully isolated.
- adds a new `useIncognito()` util for running tests in isolation.

## Testing Instructions

_Please consider any edge cases this change may have, and also other areas of the product this may impact._

1. Manually update the `tests/e2e/playwright.config.ts`:
```js
	retries: CI ? 2 : 0,
	// workers: 1, // Comment out this line
	fullyParallel: true, // Add this line
	reporter: process.env.CI
```
2. Run this command `npm run test:e2e tests/e2e/tests/mini-cart/mini-cart.block_theme.spec.ts`.
3. See the tests run in parallel and passing:

```sh
% npm run test:e2e tests/e2e/tests/mini-cart/mini-cart.block_theme.spec.ts

> @woocommerce/block-library@11.0.0-dev test:e2e
> sh ./bin/check-env.sh && npx playwright test --config=tests/e2e/playwright.config.ts

Customer state file deleted successfully.
Logged-in as customer successfully.

Running 4 tests using 4 workers

  ✓  1 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:57:6 › Mini Cart Block › should open the filled cart drawer (5.8s)
  ✓  2 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:23:6 › Mini Cart Block › should close the drawer when clicking on the close button (5.4s)
  ✓  3 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:38:6 › Mini Cart Block › should close the drawer when clicking outside the drawer (7.2s)
  ✓  4 [blockTheme] › mini-cart/mini-cart.block_theme.spec.ts:14:6 › Mini Cart Block › should open the empty cart drawer (3.8s)

  4 passed (7.6s)
```

## WooCommerce Visibility

Required:

* [ ] WooCommerce Core
* [ ] Feature plugin
* [ ] Experimental
* [x] N/A

## Checklist

Required:
* [x] This PR has either a `[type]` label or a `[skip-changelog]` label.
* [ ] This PR is assigned to a milestone.

Conditional:
* [ ] This PR has a changelog description (if `[skip-changelog]` label is not present).
* [ ] This PR adds/removes a feature flag & I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR adds/removes an experimental interfaces, and I've updated [this doc](https://github.com/woocommerce/woocommerce-blocks/blob/trunk/docs/internal-developers/blocks/feature-flags-and-experimental-interfaces.md).
* [ ] This PR has been accessibility tested.
* [ ] This PR has had any necessary documentation added/updated.

## Changelog

> Add suggested changelog entry here.

### CODE REVIEW

**Review by nielslange — APPROVED** [review]
Thanks for working on this, @dinhtungdu! I've successfully tested this change and noticed that the tests are superfast when being executed in parallel. Well done! 👏 I hope we can soon activate the parallel execution of all our E2E tests (except the side-effect-related ones). 😀

**Review by opr — CHANGES_REQUESTED** [review]
The changes to the mini cart test look great, but not super sure about the added utils. We're just re-running tests on #10561 so when they pass, I'll merge it, maybe you can rebase once that's merged and make use of the auth states?

File: tests/e2e/utils/use-incognito.ts
Comment (opr) [inline tests/e2e/utils/use-incognito.ts]: I don't think this abstraction is really necessary. We're adding functions for something that can be done in a single line anyway.

`test. use( storageState: { cookies: [], origins: [] } );`

Also, the naming implies it's opening an incognito window which may confuse folks in the future, and also since it's using `test.use` it can only be executed at the `describe` level, not within a test itself, so it may cause further confusion.

I think we should drop this.

Moreover, in https://github.com/woocommerce/woocommerce-blocks/pull/10561 we are adding auth states, so we can instead do: `test.use( { storageState: guestFile } );` to get a guest session.

Comment (opr) [reply tests/e2e/utils/use-incognito.ts]: Hmm, @dinhtungdu looks like we're not going to get #10561 merged today it seems there's some more work to do on it, so I'd be fine with you skipping waiting for that and just getting it working.

Comment (dinhtungdu) [reply tests/e2e/utils/use-incognito.ts]: You're right. I removed the utility and added comment explaining the reason.

About the using the guest file, I think it will prevent tests run in isolation because we're reusing the context. That's why I use an empty object for the state (and was over optimized by using a function to ensure the new object every time). I will verify this again when the @tarhi-saad PR is merged.

As mentioned in the code comment, I think tests should run in isolation by default, so we shouldn't use the `storageState` file in the main config. We can revisit this topic later when tests can be run in isolation are ready for isolation.

**Review by opr — APPROVED** [review]
Code changes look good! Thanks for taking my feedback into account.

The markdown link checking is due to rate limiting I think, so happy to merge even with that failing.

>  I think tests should run in isolation by default, so we shouldn't use the storageState file in the main config

You mean in `playwright.config.js` the `STORAGE_STATE` there? Yea I agree on that.

**Pull request merged**
